src/main/java/org/broadinstitute/hellbender/tools/HaplotypeCallerSpark.java

```diff
@@ -0,0 +1,329 @@
+        public int minAssemblyRegionSize = HaplotypeCaller.DEFAULT_MIN_ASSEMBLY_REGION_SIZE;
+
+        @Argument(fullName = "maxAssemblyRegionSize", shortName = "maxAssemblyRegionSize", doc = "Maximum size of an assembly region", optional = true)
+        public int maxAssemblyRegionSize = HaplotypeCaller.DEFAULT_MAX_ASSEMBLY_REGION_SIZE;
+
+        @Argument(fullName = "assemblyRegionPadding", shortName = "assemblyRegionPadding", doc = "Amount of additional bases of context to include around each assembly region", optional = true)
+        public int  assemblyRegionPadding = HaplotypeCaller.DEFAULT_ASSEMBLY_REGION_PADDING;
+
+        @Advanced
+        @Argument(fullName = "activeProbabilityThreshold", shortName = "activeProbabilityThreshold", doc="Minimum probability for a locus to be considered active.", optional = true)
+        public double activeProbThreshold = HaplotypeCaller.DEFAULT_ACTIVE_PROB_THRESHOLD;
+
+        @Advanced
+        @Argument(fullName = "maxProbPropagationDistance", shortName = "maxProbPropagationDistance", doc="Upper limit on how many bases away probability mass can be moved around when calculating the boundaries between active and inactive assembly regions", optional = true)
+        public int maxProbPropagationDistance = HaplotypeCaller.DEFAULT_MAX_PROB_PROPAGATION_DISTANCE;
+
+    }
+
+    @ArgumentCollection
+    public HaplotypeCallerArgumentCollection hcArgs = new HaplotypeCallerArgumentCollection();
+
+    @Override
+    public boolean requiresReads(){
+        return true;
+    }
+
+    @Override
+    public boolean requiresReference(){
+        return true;
+    }
+
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        final List<SimpleInterval> intervals = hasIntervals() ? getIntervals() : IntervalUtils.getAllIntervalsForReference(getHeaderForReads().getSequenceDictionary());
+        final JavaRDD<VariantContext> variants = callVariantsWithHaplotypeCaller(getAuthHolder(), ctx, getReads(), getHeaderForReads(), getReference(), intervals, hcArgs, shardingArgs);
+        writeVariants(variants);
+    }
+
+    @Override
+    public ReadFilter makeReadFilter() {
+        return HaplotypeCallerEngine.makeStandardHCReadFilter(hcArgs, getHeaderForReads());
+    }
+
+    /**
+     * Call Variants using HaplotypeCaller on Spark and return an RDD of  {@link VariantContext}
+     * @param authHolder authorization needed for the reading the reference
+     * @param ctx the spark context
+     * @param reads the reads variants should be called from
+     * @param header the header that goes with the reads
+     * @param reference the reference to use when calling
+     * @param intervals the intervals to restrict calling to
+     * @param hcArgs haplotype caller arguments
+     * @param shardingArgs arguments to control how the assembly regions are sharded
+     * @return an RDD of Variants
+     */
+    public static JavaRDD<VariantContext> callVariantsWithHaplotypeCaller(
```

Comment:
It's public and static so that It's easy to use haplotype caller as part of a longer pipeline.  I.e. you can just call this on the results of ApplyBQSR and get variants out.  I can remove it / change it if you think there is a different better way to make it modular.  Or add a comment to make it more obvious what it's for.